[EMCAL-688] Add exception for uninitilised lookup table

mhemmer-cern opened the pull request:
- Add exception class to throw when trying to determine exotic clusters but the neeed look up table is not initilised
	- The class UninitLookUpTableException is neseted within the cluster factory class
	- throw will happen inside of setIsExotic function which is called inside buildCluster function
	- throw will only happen if ecell < mExoticCellMinAmplitude is false
- Add first step to merge set up function for the cluster, cell and index container together with the look up table
	- new setContainer function expects all three objects as span and will set them
	- next step would be to change the current calls of the three (four) setter everywhere with this one

mhemmer-cern committed:
[EMCAL-688] Add exception for uninitilised lookup table
- Add exception class to throw when trying to determine exotic clusters but the neeed look up table is not initilised
	- The class UninitLookUpTableException is neseted within the cluster factory class
	- throw will happen inside of setIsExotic function which is called inside buildCluster function
	- throw will only happen if ecell < mExoticCellMinAmplitude is false
- Add first step to merge set up function for the cluster, cell and index container together with the look up table
	- new setContainer function expects all three objects as span and will set them
	- next step would be to change the current calls of the three (four) setter everywhere with this one

mfasDa commented on Detectors/EMCAL/base/include/EMCALBase/ClusterFactory.h:
```diff
@@ -311,6 +311,9 @@ class ClusterFactory
   Double_t tMaxInCm(const Double_t e = 0.0, const int key = 0) const;
 
+  bool getLookUpInit() const { return mLookUpInit; }
+  void setLookUpInit(bool lookUpInit) { mLookUpInit = lookUpInit; }
```
mLookupInit is a state variable, I would not expose it to the world.

mhemmer-cern replied:
Done!

mfasDa reviewed (changes requested):
Please protect the state variable mLookupTable from outside access (remove `setLookupInit`), otherwise it is good to go.

mhemmer-cern committed:
[EMCAL-688] Remove setLookUpInit to protect mLookUpInit from outside
- Remove setLookUpInit as suggested by Markus to protect member variable mLookUpInit to be changed from outside

mfasDa reviewed (approved):
Looks good now!

alibuild commented:
Error while checking build/O2/fullCI for 8aa2731fd952d1414537e1093188eccc5e98c2d6 at 2023-03-30 22:50:
```
## sw/BUILD/O2-full-system-test-latest/log
command /sw/slc8_x86-64/O2/11062-slc8_x86-64-local1/prodtests/full-system-test/dpl-workflow.sh had nonzero exit code 128
[5908:BadMapCalibSpec]: [20:38:22][ERROR] Insufficient statistics: 3 entries in lowE histo, do nothing
[ERROR] pid 5891 (gpu-reconstruction) crashed with 128
[ERROR] pid 5964 (primary-vertexing) crashed with 128
[ERROR] pid 5965 (tpc-track-interpolation) crashed with 128
[ERROR] pid 5966 (mch-cluster-transformer) crashed with 128
[ERROR] pid 5981 (qc-task-ITS-ITSTrackTask) crashed with 128
[ERROR] pid 5949 (tpc-calibrator-gainmap-tracks) crashed with 128
[ERROR] SEVERE: Device gpu-reconstruction (5891) returned with 128


## sw/BUILD/o2checkcode-latest/log
--
++ echo '========== List of errors found =========='
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/3be6ce408646ee1b3e6d98f59d0dda9dc00001a2/slc8_x86-64/o2checkcode/1.0-local337/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/11062/8aa2731fd952d1414537e1093188eccc5e98c2d6/build_O2_fullCI/pretty.html).
